src/api/java/mcp/mobius/waila/api/component/NamedItemListComponent.java

```diff
@@ -0,0 +1,66 @@
+package mcp.mobius.waila.api.component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import mcp.mobius.waila.api.ITooltipComponent;
+import mcp.mobius.waila.api.__internal__.ApiSide;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.ItemStack;
+
+/**
+ * Component that renders items with their names.
+ */
+@ApiSide.ClientOnly
+public class NamedItemListComponent implements ITooltipComponent {
+
+    private static final NamedItemComponent MORE = new NamedItemComponent(ItemStack.EMPTY.setHoverName(Component.literal("...")));;
```

Comment:
```suggestion
    private static final NamedItemComponent MORE = new NamedItemComponent(ItemStack.EMPTY.copy().setHoverName(Component.literal("...")));;
```
Don't modify the `EMPTY` stack directly, copy it first. Though I think drawing the text manually would've been the better approach than this.

Reply:
Yeah, my bad. Removed this completely, `ctx.drawString` call to show three dots works just fine